### PR TITLE
feat: Added ability to use ASK custom build hooks for local package linking

### PR DESCRIPTION
**Pull request by anfazio**
# Added ability to use ASK custom build hooks for local package linking

## Description

Added 'ask-hooks-directory' property to abcConfig.json to specify a directory containing build hooks for the ASK CLI. If this property is present, then the contents of the given directory will be copied into the `.deploy/story-id/hooks` directory during staging. Also added an example `build.sh` that uses `yarn` instead of `npm` and added instructions to the top-level README describing how to link local packages.

## Motivation and Context

The SFB repo relies on `yarn` to handle local package dependencies, this makes it easier to test and use modifications made to the SFB packages locally.

## Testing

Deployed a sample skill with local package modifications to multiple stages, ran the skill stages and verified that altered behavior was as expected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

[issues]: https://github.com/alexa-games/skill-flow-builder/issues

### CODE REVIEW

File: scripts/local-packages/hooks/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#   http://aws.amazon.com/asl/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+set -euo pipefail
+
+yarn install --production --quite
```
Comment (anfazio) [inline scripts/local-packages/hooks/build.sh:20]: Might make sense to change this to `npx yarn` to be consistent with our other scripts

File: scripts/local-packages/hooks/build.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+zip -r $1 *
```
Comment (tyliec) [inline scripts/local-packages/hooks/build.sh:21]: Might consider a check for Windows/Mac in order for 7z vs. zip here

File: scripts/local-packages/hooks/build.sh (hunk 1 above)
Comment (tyliec) [reply scripts/local-packages/hooks/build.sh:20]: I've never seen a `--quite` flag before, what does it do? ._.

File: packages/sfb-cli/lib/specialPaths.ts [hunk 3]
```diff
@@ -161,4 +162,10 @@ export class SpecialPaths {
             ? pathModule.join(configPaths.askSkillFullPath, 'models')
             : pathModule.join(configPaths.skillPackagePath, 'interactionModels', 'custom');
     }
+
+    public static getAskLambdaCodeDeployPath(configPaths: ConfigPaths) {
+        return SpecialPaths.isAskCliV1(configPaths)
+        ? SpecialPaths.getLambdaCodeDeployPath(configPaths)
+        : pathModule.join(configPaths.askSkillFullPath, ASK_METADATA_DIRECTORY, 'lambda')
```
Comment (tyliec) [inline packages/sfb-cli/lib/specialPaths.ts:169]: nit: missing semi

File: scripts/local-packages/hooks/build.sh (hunk 1 above)
Comment (anfazio) [reply scripts/local-packages/hooks/build.sh:20]: It's a misspelling of quiet that appears to be accepted, that said, I don't think it's necessary in our case and I think I might remove it.

File: scripts/local-packages/hooks/build.sh (hunk 2 above)
Comment (anfazio) [reply scripts/local-packages/hooks/build.sh:21]: Actually, there's a .ps1 script that needs to be included for Windows, will check that in as well

File: packages/sfb-cli/lib/stageCommand.ts [hunk 4]
```diff
@@ -219,8 +221,19 @@ export class StageCommand implements Command {
         await FileUtils.recursiveCopy(pathModule.join(dirs.buildOutputPath, '*'), lambdaCodeDeployPath, { makeDestinationWritable: true });
 
         await FileUtils.recursiveCopy(pathModule.join(dirs.codeBuildOutputPath, '*'), lambdaCodeDeployPath, { makeDestinationWritable: true });
+        
+        await this.adjustLocalDependencyPaths(dirs, configDirs, lambdaCodeDeployPath);
 
-        await this.setupNodeModulesUsingInstallProduction(dirs, lambdaCodeDeployPath, configHelper.getValue("sfbLocalTesting", undefined, "en-US"));
+        const hooksDestPath = FileUtils.fixpath(pathModule.join(configDirs.askSkillFullPath, HOOKS_DIRECTORY))
```
Comment (tyliec) [inline packages/sfb-cli/lib/stageCommand.ts:227]: nit: missing semicolon

**Review by tyliec — APPROVED** [review]

File: packages/sfb-cli/lib/stageCommand.ts [hunk 5]
```diff
@@ -249,56 +262,43 @@ export class StageCommand implements Command {
...
+
+        this.adjustDependencyRelativePaths(dirs, askLambdaCodeDeployPath, packageManifest.dependencies);
+        this.adjustDependencyRelativePaths(dirs, askLambdaCodeDeployPath, packageManifest.devDependencies);
+
+        const lambdaPackageManifestFilePath = pathModule.join(lambdaCodeDeployPath, PACKAGE_MANIFEST_FILE);
+        fs.writeFileSync(lambdaPackageManifestFilePath, JSON.stringify(packageManifest, null, 4));
+    }
+
+    private adjustDependencyRelativePaths(dirs: SpecialPaths, lambdaCodeDeployPath: string, dependencies: any): void {
+        for (const [name, path] of Object.entries(dependencies as Record<string, string>)) {
```
Comment (tyliec) [inline packages/sfb-cli/lib/stageCommand.ts:292]: This line is way too cool for me to not comment on how cool it is!

**Pull request merged**